doc/CONTRIBUTE.md

```diff
@@ -5,93 +5,67 @@
 Directory architecture :
 
 * app.js: specific settings and actually start the server
-* index.js: contain the generic server implementation
-* lib/: contain the classes and functions used in the plugins
-  * serverPlugins/: server plugins that do things general to the server, 
-  properties and method are added to the server object in them
-  * playerPlugins/: player plugins that do things for each player, properties and method are added to the player object in them
+* dist/: Contains "compiled" code (go to current directory in console and type `gulp` to generate this)
+* src/: Source files for the project
+* src/index.js: contain the generic server implementation
+* src/lib: contain the classes and functions used in the plugins
+  * plugins/: All of the default plugins made to simulate vanilla
+  * worldGeneraions/: Contains default world generations, however plugins can use their own
   
-Structure of a server plugin:
+## Structure of a plugin
 
 ```js
-module.exports=inject;
 
-function inject(serv)
-{
-  // add methods and properties to serv
-}
-```
+// Each of these are called an "inject" because you're injecting properties, events, methods, or data into the objects
 
-Structure of a player plugin :
+module.exports.server = function(serv) { // Create your server events here
+  serv.spawnPoint = ...;
+  serv.on('...', ...);
+}
 
-```js
-module.exports=inject;
+module.exports.entity = function(entity, serv) { // Called whenever an entity is created, do NOT do serv.on here
+  entity.health = 10; // Start with 10 health out of 20
+  entity.on('...', ...);
+  // serv.on('...', ...); NOOOO
+}
 
-function inject(serv,player)
-{
-  // add methods and properties to player
-  // you can use serv, but you shouldn't add things to it here
+module.exports.player = function(player, serv) { // Player is a type of entity (entity inject is called first) with added properties and functions
+  player.setXp(100); // Example of a property player entities have but not other entities
+  player.on(',,,', ...);
+  // serv.on('...', .– don't even think about it
 }
+
 ```
 
 ## Logs and event
 
 In order to keep logging independent from the rest of the server and to let people react in other ways than logging,
-server and player events should be emitting and the logging should only take place in response to these events
```

Comment:
yes, but what I meant is player and server events are emitted, and then listened to in log.js which call serv.log